CONTRIBUTING.md

```diff
@@ -1,17 +1,183 @@
-chat](https://discord.gg/xS7Z362), or #llvm IRC channel on
-[OFTC](https://oftc.net/).
+The Arm Toolchain repository is synchronized with the upstream
+*llvm-project* including the main and release branches. Arm Toolchain
+for Embedded aims to stay close to the head of the *picolibc* main
+branch. The *newlib* overlay package aims to track the most recent
+*newlib* release.
 
-The LLVM project has adopted a [code of conduct](https://llvm.org/docs/CodeOfConduct.html) for
-participants to all modes of communication within the project.
+All changes that can be made in the upstream project, must be made in
+the upstream project. Please see the [Downstream Patch Policy](#downstream-patch-policy)
+below for what Arm considers a downstream patch along with the
+additional requirements for such a patch.
+
+For guidance on how to contribute to the upstream projects see:
+* *llvm-project* [Contributing to LLVM](https://llvm.org/docs/Contributing.html).
+* *picolibc* GitHub.
+[Pull requests documentation](https://docs.github.com/en/pull-requests)
+
+## Ways to contribute
+
+### Report an issue
+
+Please make sure that the issue is specific to the *arm-toolchain*
+before reporting. If the issue exists in *llvm-project* or *picolibc*
+then please report the issue in the upstream project.
+
+Please create a GitHub issue in the *arm-toolchain* project
+[Issues](https://github.com/arm/arm-toolchain/issues)
+list.
+
+### Submit a fix
+
+For a small change, please create a Pull Request as described in
+_How to submit a change_ section below.
+
+### Suggest a feature or bigger change
+
+For a bigger change, please create an issue in the
+*arm-toolchain* project
+[Issues](https://github.com/arm/arm-toolchain/issues)
+list and put [RFC] in the subject line (Request for Comments) to initiate the discussion
+first, before submitting a PR for the change.
+
+There is no formal template for an `rfc`, however it would be good to
+explain the purpose of the change and the key design options, proposed
+decisions.
+
+## How to submit a change
+
+Contributions are accepted under the
+[Apache License 2.0 with LLVM Exceptions](https://github.com/arm/arm-toolchain/blob/arm-software/LICENSE.TXT).
+Only submit contributions where you have authored all of the code.
+
+### Pull request
+
+This project follows the conventional
+[GitHub pull request](https://docs.github.com/en/pull-requests) flow.
+
+### Testing
+
+Please ensure your change doesn't break tests. (The project doesn't yet have
+GitHub actions to do this). Except for documentation changes, please check that
+this passes:
+
+```
+ninja check-llvm-toolchain
+```
+
+### Coding style
+
+Use the following commands to check the scripts before submitting a pull
+request:
+
+```
+$ ./setup.sh
+$ ./run-precommit-checks.sh
+```
+
+## Downstream Patch Policy
+
+### What is considered a downstream patch?
+
+A patch is considered a downstream patch if:
+
+* It touches any part of the *llvm-project* repository outside the
```

Comment:
Should this be *arm-toolchain* repository? Or maybe more complicated statement like llvm-project files inside arm-toolchin repo?

Reply:
I've used the *arm-toolchain* repository and tweaked the language a bit.